File: C-Unexpressed.cpp
#include <iostream>
#include <vector>
#include <set>
using namespace std;

int main(){
    long long n;
    cin >> n;
    vector<long long> memo(0);
    long long a, cu;
    a = sqrt(n);
    
    /*cout << a << endl;*/
    for (int i = 2; i < a + 1; i ++){
        cu = i;
        while(cu * i <= n){

            cu *= i;
            memo.push_back(cu);
            
            /*cout << cu << endl;*/
        }
        
    }
    set<long long> S(memo.begin(), memo.end());
    vector<long long> mome(S.begin(), S.end());

    cout << n - mome.size() << endl;
}

File: C-Unexpressed.py
import math
n = int(input())
memo = [0]
n_sqrt = int(math.sqrt(n)) + 1
for i in range(2, n_sqrt):
    x = i
    while(x * i <= n):
        x *= i
        memo.append(x)
    
memo = set(memo)
memo = list(memo)

ans = n - len(memo) + 1

print(ans)




